# 4.6 Combining & Exporting Data

In [1]:
# Import Libraries

import pandas as pd
import numpy as np
import os

## Import df_ords_prior

In [2]:
# Import df_ords_prior

path = r'C:\Users\natha\OneDrive\Desktop\Data Analytics\Jupyter\09-2023 Instacart Basket Analysis'
df_ords_prior = pd.read_csv(os.path.join(path, '02 Data', 'Original Data', 'orders_products_prior.csv'), index_col = False)

In [3]:
# Check output

df_ords_prior.head()

order_id  product_id  add_to_cart_order  reordered
0         2       33120                  1          1
1         2       28985                  2          1
2         2        9327                  3          0
3         2       45918                  4          1
4         2       30035                  5          0

In [4]:
# Check shape

df_ords_prior.shape

(32434489, 4)

## Import df_ords

In [5]:
# Import df_ords

path = r'C:\Users\natha\OneDrive\Desktop\Data Analytics\Jupyter\09-2023 Instacart Basket Analysis'
df_ords = pd.read_csv(os.path.join(path, '02 Data', 'Prepared Data', 'orders_checked.csv'), index_col = False)

In [6]:
# Check output

df_ords.head()

order_id  user_id  order_number  orders_day_of_week  order_hour_of_day  \
0   2539329        1             1                   2                  8   
1   2398795        1             2                   3                  7   
2    473747        1             3                   3                 12   
3   2254736        1             4                   4                  7   
4    431534        1             5                   4                 15   

   days_since_prior_order  is_first_order  
0                     NaN               1  
1                    15.0               0  
2                    21.0               0  
3                    29.0               0  
4                    28.0               0

In [7]:
# Check shape

df_ords.shape

(3421083, 7)

# Merge df_ords and df_ords_prior based on common column 'order_id' (INNER JOIN)

In [8]:
# Merge df_ords and df_ords_prior based on common column 'order_id' as the 'key'

df_merged_large = df_ords.merge(df_ords_prior, on = 'order_id', indicator = True)

In [9]:
# check merge flag frequency (inner joins) [inner joins labelled as 'both']

df_merged_large['_merge'].value_counts()

_merge
both          32434489
left_only            0
right_only           0
Name: count, dtype: int64

In [10]:
# check merged dataframe (inner joins)

df_merged_large.head()

order_id  user_id  order_number  orders_day_of_week  order_hour_of_day  \
0   2539329        1             1                   2                  8   
1   2539329        1             1                   2                  8   
2   2539329        1             1                   2                  8   
3   2539329        1             1                   2                  8   
4   2539329        1             1                   2                  8   

   days_since_prior_order  is_first_order  product_id  add_to_cart_order  \
0                     NaN               1         196                  1   
1                     NaN               1       14084                  2   
2                     NaN               1       12427                  3   
3                     NaN               1       26088                  4   
4                     NaN               1       26405                  5   

   reordered _merge  
0          0   both  
1          0   both  
2          0   both  
3          0   both  
4          0   both

In [14]:
# check merged dataframe shape

df_merged_large.shape

(32434489, 11)

### *Inner Join merge is what we want for our Instacart Project, as we want merges with a full match for our purposes. In this case we have 32434489 rows with full merges.*

# Exporting Data to CSV and PKL files

In [11]:
# Export dataframe to CSV

df_merged_large.to_csv(os.path.join(path, '02 Data','Prepared Data', 'orders_products_combined.csv'), index=False)

In [12]:
# Export dataframe to PKL

df_merged_large.to_pickle(os.path.join(path, '02 Data','Prepared Data', 'orders_products_combined.pkl'))